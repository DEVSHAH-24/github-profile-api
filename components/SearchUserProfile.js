import React, { Component } from "react";
import {
  StyleSheet,
  Text,
  View,
  Button,
  TouchableOpacity,
  TouchableHighlight,
} from "react-native";
import { useEffect, useState } from "react";
import SearchBarFcn from "./SearchBar";
import { Searchbar } from "react-native-paper";
import UserInfoCard from "./Info";
import axios from "axios";
import PublicRepos from "./publicRepos";
//import { SearchBar } from 'react-native-elements';

import SearchBar from "react-native-searchbar";
import { signOut } from "firebase/auth";
import { useNavigation } from "@react-navigation/native";
import { auth } from "../firebase";

function Main() {
  const [userData, setUserData] = useState({});

  const [currentName, setCurrentName] = useState("");
  var gitHubUrl = `https://api.github.com/users/${currentName}`;

  const getUserData = async () => {
    // axios.get(gitHubUrl).then((response)=> {
    //   response.message !== "Not Found" ? setUserData(response) : console.log
    // })
    const response = await fetch(gitHubUrl);
    const jsonData = await response.json();
    if (jsonData && jsonData.message !== "Not Found") {
      setUserData(jsonData);
      console.log(jsonData);
    } else if (username !== "") {
      console.log("Username does not exist");
      setUserData({});
    } else {
      setUserData({});
    }
  };
  // useEffect(() => {
  // }, [username]);
  // const onChangeSearch = query => setUsername(query);
  const navigation = useNavigation();
  const handleSignOut=()=> {
    signOut(auth).then(()=>{
      console.log('signingout')
      navigation.navigate('Login');
    }).catch(e => alert(e.message))
  }


  return (
    <View style={styles.container}>
      <View style={styles.searchLogout}>
        <Searchbar
          placeholder="Search"
          onChangeText={(e) => setCurrentName(e)}
          onIconPress={() => getUserData()}
        />
        <Button title="Logout" onPress={handleSignOut}></Button>
      </View>
      <View style={styles.container}>
        <UserInfoCard userData={userData} />
      </View>

      <View style={styles.container}>
        <Text style={styles.titleStyle}>Public repos</Text>
        <PublicRepos userData={userData} />
      </View>
    </View>
  );
}
const styles = StyleSheet.create({
  // outerContainer: {
  //   backgroundColor: "grey",
  // },
  container: {
    //  flexDirection: "horizontal",
    alignItems: "center",
  },
  searchLogout: {
    flexDirection: "row",
    alignItems: "flex-end",
    padding: 20,
    alignContent: "space-between",
  },
  titleStyle: {
    fontSize: 20,
    fontWeight: "500",
    paddingVertical: 20,
    color: "green",
  },
});
export default Main;
