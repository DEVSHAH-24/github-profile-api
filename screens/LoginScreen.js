import {
  KeyboardAvoidingView,
  StyleSheet,
  Text,
  View,
  TextInput,
} from "react-native";
import React, { useEffect, useState } from "react";
import { TouchableOpacity } from "react-native-web";
import { auth } from "../firebase";
import {
  getAuth,
  createUserWithEmailAndPassword,
  onAuthStateChanged,
  signInWithEmailAndPassword,
} from "firebase/auth";
import { NavigationContainer, useNavigation } from "@react-navigation/native";

const LoginScreen = () => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const navigation = useNavigation();

  const handleSignUp = () => {
    createUserWithEmailAndPassword(auth, email, password)
      .then((userCredentials) => {
        const user = userCredentials.user;
        console.log(user.email);
      })
      .catch((e) => alert(e.message));
  };

  const handleLogin = () => {
    
    signInWithEmailAndPassword(auth, email, password)
      .then((userCredentials) => {
        const user = userCredentials.user;
        console.log(user.email);
        navigation.navigate("Home");

      })
      .catch((e) => alert(e.message));
  };
  useEffect(() => {
    const unsubscribe = onAuthStateChanged(auth, (user) => {
      if (user) {
        navigation.navigate("Home");
      }
    });
    return unsubscribe;
  }, []);

  return (
    <KeyboardAvoidingView style={styles.container} behavior="padding">
      <Text style={styles.textStyle}>Welcome!</Text>
      <View style={styles.inputContainer}>
        <TextInput
          placeholder="Email"
          value={email}
          onChangeText={(text) => setEmail(text)}
          style={styles.input}
        />
        <TextInput
          placeholder="Password"
          secureTextEntry
          style={styles.input}
          value={password}
          onChangeText={(text) => setPassword(text)}
        />
      </View>
      <View style={styles.buttonContainer}>
        <TouchableOpacity
          onPress={
            handleLogin
          }
          style={styles.button}
        >
          <Text style={[styles.button, styles.buttonOutline]}>Login</Text>
        </TouchableOpacity>

        <TouchableOpacity onPress={handleSignUp} style={styles.button}>
          <Text style={[styles.button, styles.buttonOutline]}>Register</Text>
        </TouchableOpacity>
      </View>
    </KeyboardAvoidingView>
  );
};

const styles = StyleSheet.create({
  textStyle: {
    fontSize: 25,
    fontWeight: "700",
    paddingVertical: 20,
  },

  container: {
    justifyContent: "center",
    alignItems: "center",
    flex: 1,
  },
  inputContainer: {
    width: "80%",
    
  },
  input: {
    backgroundColor: "white",
    paddingHorizontal: 15,
    paddingVertical: 2,
    borderRadius: 10,
    marginTop: 5,
  },
  button: {
    backgroundColor: "#0782F9",
    width: "50%",
    padding: 15,
    borderRadius: 10,
    alignItems: "center",
    justifyContent:'center',
    textAlign:'center'
  },
  buttonContainer: {
    width: "60%",
    justifyContent: "center",
    alignItems: "center",
    marginTop: 40,
  },
  buttonOutline: {
    backgroundColor: "white",
    marginTop: 5,
    borderColor: "#0782F9",
    borderWidth: 2,
  },
  buttonText: {
    color: "white",
    fontWeight: "700",
    fontSize: 16,
    alignItems: 'center'
  },
});
export default LoginScreen;
