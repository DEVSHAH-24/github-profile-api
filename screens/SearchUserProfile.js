import React from "react";
import { useState } from "react";
import { StyleSheet, Text, View } from "react-native";

import { Searchbar } from "react-native-paper";
import { appStyles } from "../styles/appStyles";
import PublicRepos from "../components/PublicRepos";
import UserInfoCard from "../components/UserInfoCard";

const Main = () => {
  const [userData, setUserData] = useState({});

  const [currentName, setCurrentName] = useState("");
  var gitHubUrl = `https://api.github.com/users/${currentName}`;

  const getUserData = async () => {
    try {
      if (currentName !== "") {
        const response = await fetch(gitHubUrl);
        const jsonData = await response.json();
        if (jsonData && jsonData.message !== "Not Found") {
          setUserData(jsonData);
          console.log(jsonData);
        } else if (jsonData.message === "Not found") {
          alert("Username not found", "Please try a different username");

          console.log("Username does not exist");
          setUserData({});
        } else {
          alert("Username not found", "Please try a different username");
          setUserData({});
        }
      }
    } catch (error) {
      alert(error);
      console.log(error, "error msg");
    }
  };

  return (
    <View style={appStyles.container}>
      <View style={appStyles.searchLogout}>
        <Searchbar
          placeholder="Search"
          onChangeText={(e) => setCurrentName(e)}
          onIconPress={() => getUserData()}
        />
      </View>
      <View>
        <UserInfoCard userData={userData} />
      </View>

      <PublicRepos userData={userData} />
    </View>
  );
};

export default Main;
