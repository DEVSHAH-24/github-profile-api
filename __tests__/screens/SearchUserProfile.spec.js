import { shallow } from "enzyme";
import * as React from "react";
import TestRenderer, { act, create } from "react-test-renderer";
import { Main } from "../../screens/SearchUserProfile";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-17-updated";

Enzyme.configure({ adapter: new Adapter() });

const getComponent = (props = {}) => shallow(<Main {...props} />);
const mockedUsedNavigate = jest.fn();

jest.mock("@react-navigation/native", () => {
  const actualNav = jest.requireActual("@react-navigation/native");
  return {
    ...actualNav,

    useNavigation: () => ({
      navigation: mockedUsedNavigate,
    }),
  };
});

describe("SearchUserProfile - match snapshot", () => {
  beforeEach(() => {
    mockedUsedNavigate.mockReset();
  });
  it("Snapshot", () => {
    const component = getComponent();

    const testRenderer = TestRenderer.create(component);
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it("Search input", () => {
    const component = getComponent();
    const tree = create(component);
    const setFn = jest.fn();
    const searchBar = tree.root.findByProps({ testID: "searchbar" }).props;

    act(() => {
      //searchBar.props({ onChangeText: { setFn } });
    });
    expect(setFn).toHaveBeenCalled();
  });

  it("on change text", () => {
    const component = getComponent();
    console.log(component.debug());
    //component.find('[testID="searchbar"]').prop("OnIconPress");
    expect(component).toMatchSnapshot();
  });
});
