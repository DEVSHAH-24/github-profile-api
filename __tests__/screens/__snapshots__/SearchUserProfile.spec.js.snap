// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchUserProfile - match snapshot Search input 1`] = `<Main />`;

exports[`SearchUserProfile - match snapshot Snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "paddingHorizontal": 5,
    }
  }
>
  <View
    style={
      Object {
        "alignContent": "space-between",
        "alignItems": "space-between",
        "flexDirection": "row",
        "padding": 10,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderRadius": 4,
          "elevation": 4,
          "flexDirection": "row",
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 3,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 4,
        }
      }
    >
      <View
        accessibilityLabel="search"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "margin": 6,
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 18,
                "height": 36,
                "width": 36,
              },
              undefined,
              undefined,
            ],
          ]
        }
      >
        <View>
          <Text />
        </View>
      </View>
      <TextInput
        accessibilityRole="search"
        accessibilityTraits="search"
        keyboardAppearance="light"
        onChangeText={[Function]}
        placeholder="Search"
        placeholderTextColor="rgba(0, 0, 0, 0.54)"
        returnKeyType="search"
        selectionColor="#6200ee"
        style={
          Array [
            Object {
              "alignSelf": "stretch",
              "flex": 1,
              "fontSize": 18,
              "minWidth": 0,
              "paddingLeft": 8,
              "textAlign": "left",
            },
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            undefined,
          ]
        }
        testID="searchbar"
        underlineColorAndroid="transparent"
      />
      <View
        accessibilityLabel="clear"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "margin": 6,
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 18,
                "height": 36,
                "width": 36,
              },
              Object {
                "opacity": 0.32,
              },
              undefined,
            ],
          ]
        }
      >
        <View>
          <Text />
        </View>
      </View>
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "padding": 35,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://reactnative.dev/img/tiny_logo.png",
          }
        }
        style={
          Object {
            "borderRadius": 50,
            "height": 100,
            "width": 100,
          }
        }
      />
      <Text
        style={
          Object {
            "fontSize": 25,
            "fontWeight": "700",
            "paddingVertical": 20,
          }
        }
      >
        Name
      </Text>
      <Text
        style={
          Object {
            "fontSize": 15,
            "fontWeight": "400",
            "paddingVertical": 5,
          }
        }
      >
        Location will appear here
      </Text>
      <Text
        style={
          Object {
            "fontSize": 15,
            "fontWeight": "400",
            "paddingVertical": 5,
          }
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "paddingHorizontal": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "green",
          "fontSize": 20,
          "fontWeight": "500",
          "paddingVertical": 10,
        }
      }
    >
      Public repositories: 
    </Text>
    <RCTScrollView
      data={Object {}}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEnabled={true}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={Array []}
      viewabilityConfigCallbackPairs={Array []}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;
